src/Plugin/AuthenticationPluginManager.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful\AuthenticationPluginManager.
+ */
+
+namespace Drupal\restful\Plugin;
+
+use Drupal\Core\Plugin\DefaultPluginManager;
+use Drupal\Core\Plugin\Factory\ContainerFactory;
+use Drupal\Core\Plugin\Discovery\YamlDiscovery;
+use Drupal\plug\Util\Module;
+
+/**
+ * Authentication plugin manager.
+ */
+class AuthenticationPluginManager extends DefaultPluginManager {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaults = array(
+    // Human readable label for the authentication.
+    'label' => '',
+    // A description of the plugin.
+    'description' => '',
+    'settings' => array(),
+    'id' => '',
+  );
+
+  /**
+   * Constructs AuthenticationPluginManager.
+   *
+   * @param \Traversable $namespaces
+   *   An object that implements \Traversable which contains the root paths
+   *   keyed by the corresponding namespace to look for plugin implementations.
+   * @param \DrupalCacheInterface $cache_backend
+   *   Cache backend instance to use.
+   */
+  public function __construct(\Traversable $namespaces, \DrupalCacheInterface $cache_backend) {
+    $this->discovery = new YamlDiscovery('authentication', Module::getDirectories());
```

Comment:
I wanted to have an implementation of both so we could decide if we wanted to use YAML or annotations.

It seems clear that we both prefer annotations, so this will need to get converted. Luckily enough the only part that changes is the discovery :-)